paddle/fluid/prim/api/auto_code_generated/template/utils.cc.j2

```diff
@@ -173,3 +174,27 @@ paddle::framework::TransToProtoVarType({{src_name}})
 {%- macro sequence(lsymbol, rsymbol, delimiter, items) -%}
 {{lsymbol}}{%- for item in items -%}{{item}}{{delimiter if not loop.last else "" }}{%- endfor -%}{{rsymbol}}
 {%- endmacro -%}
+
+
+{%- macro static_prim_api_output_dtype(inputs, attrs) -%}
+  {%- set is_set = [] -%}  {#- why not use boolean, ref: https://stackoverflow.com/questions/17925674/jinja2-local-global-variable -#}
+  {%- if not is_set -%} {#- use DataType attr as default output dtype -#}
+    {%- for attr in attrs -%}
+      {%- if attr.typename is datatype -%} 
+{{attr.name}}
+        {%- do is_set.append(1) -%}
+      {%- endif -%}
+    {%- endfor -%}
+  {%- endif -%}
+  {%- if not is_set -%} {#- use first input named x dtype as default output dtype -#}
+    {%- for input in inputs -%}
+      {%- if input.typename == 'Tensor' and input.name == 'x' -%}
```

Comment:
only support x.dtype() ?

Reply:
Use x.dtype is a default behavior. should use InferVarType to ensure the dtype.